xgi/classes/function.py

```diff
@@ -903,3 +905,71 @@ def subfaces(edges, order=None):
 
         faces += faces_to_add
     return faces
+
+
+def complement(H):
+    """Returns the complement of hypergraph H.
+
+    The complement of an hypergraph will have the same nodes (same indexes) and
+    will contain every possible hyperedge linking these nodes, provided it is
+    not already present in the hypergraph and it is not larger than its largest
+    hyperedge.
+    Hyperedges of size one are taken into account.
+
+
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        Hypergraph to complement.
+
+    Returns
+    -------
+    Hc : xgi.Hypergraph
+        Complement of H.
+    """
+
+    from ..generators.classic import empty_hypergraph
+
+    N = len(H.nodes)
+    M = len(H.edges)
+
+    if N == 0:
+        return empty_hypergraph()
+    else:
+        # list the edges in H (as strings such as '001101')
+        I = incidence_matrix(H, sparse=False)
+        edges = set()
+        for row in I.transpose():
+            edge = str()
+            for _01 in row:
+                edge += str(_01)
+            edges.add(edge)
+
+        # list all possible edges of size <= max_edge_size
+        if M == 0:
+            max_edge_size = 0
+        else:  # there is at least one non-empty edge
+            max_edge_size = H.edges.order.max() + 1
+
+        possible_edges = set()
+        for size in range(1, max_edge_size + 1):
+            possible_edges = possible_edges.union(binomial_sequence(size, N))
+
+        # instanciate an hypergraph an add every node (in case a node is in no hyperedge)
+        Hc = Hypergraph()
+        for i in range(N):
+            Hc.add_node(i)
```

Comment:
This only works if H has nodes with names from 0 to N-1, but not for strings or other orders (docstring specify Hc has same nodes and indices).
What about
```suggestion
        Hc.add_nodes_from(H.nodes)
```

Reply:
This is fixed.